electron/src/chatBro/queries/TopFriends/TopFriends.ts

```diff
@@ -1,62 +1,70 @@
 import * as sqlite3 from 'sqlite3';
 import { DEFAULT_LIMIT } from '../../constants/defaultFilters';
+import getAllFilters from './filters';
+import { Columns, OutputColumns } from './columns';
 
 import * as sqlite3Wrapper from '../../../utils/initUtils/sqliteWrapper';
 
-const Columns = {
-  COUNT: 'count',
-  PHONE_NUMBER: 'phone_number',
-  FRIEND: 'friend',
-  IS_FROM_ME: 'sent',
-};
-
-const FROM_QUERY = `
-  SELECT
+const getCoreQuery = (allFilters: string) => {
+  return `SELECT
     COUNT(*) as ${Columns.COUNT},
     h.id as ${Columns.PHONE_NUMBER},
     COALESCE(h.contact_name, h.id) as ${Columns.FRIEND},
     is_from_me as ${Columns.IS_FROM_ME}
   FROM message
     JOIN handle h
-    ON h.ROWID = handle_id
-    -- NOTE: Excludes group chats
-    WHERE message.cache_roomnames IS NULL
+    ON
+      h.ROWID = handle_id
+    -- NOTE: filters should always be applied as earliest as possible
+    ${allFilters}
   GROUP BY h.id, is_from_me
 `;
+};
 
-function getSentOrRecieved(alias: string, isFromMeToggle: string) {
+function getSentOrRecieved(
+  alias: string,
+  isFromMeToggle: string,
```

Comment:
Think it'd be better to have isFromMeToggle just be isFromMe and as a boolean, and convert to the string within the function. Seems kinda awk to explicitly pass in '0' as the user

Reply:
Fair enough. Thanks, changed.